tests/Dockerfile

```diff
@@ -0,0 +1,36 @@
+FROM rust:latest
+
+# Install Node.js
+ENV NODE_VERSION 12.0.0
+RUN curl https://raw.githubusercontent.com/creationix/nvm/master/install.sh | bash \
+  && . $HOME/.nvm/nvm.sh \
+  && nvm install $NODE_VERSION \
+  && nvm alias default $NODE_VERSION \
+  && nvm use default
+ENV PATH /root/.nvm/versions/node/v$NODE_VERSION/bin:$PATH
+# Check that Node.js was correctly installed
+RUN node --version 
+
+# Create blank project
+RUN USER=root cargo new --bin starship
+WORKDIR /starship
+
+# We want dependencies cached, so copy those first
+COPY ./Cargo.lock ./Cargo.lock 
+COPY ./Cargo.toml ./Cargo.toml 
+
+# Cargo.toml will fail to parse without my_benchmark
+RUN mkdir benches
+RUN touch benches/my_benchmark.rs
+
+# This is a dummy build to get dependencies cached
+RUN cargo build --release
```

Comment:
If we wanna go this route it could potentially useful to do multi stage builds or alternatively just build the base image with all the environments and then push that image to docker-hub or the azure pipeline image repo. 

We could then reuse that image, could make things a lot faster since its reusing cached images. 
Especially since they gonna pretty big with all the environments needed to run the tests.

Reply:
> it could potentially useful to do multi stage builds

I haven't done multi-stage builds before, but I'm open to it.

> alternatively just build the base image with all the environments and then push that image to docker-hub or the azure pipeline image repo.

Yep, that's what I had in mind.
For the time being the base image will be changing _very_ often, so I wasn't sure how we would want to manage it so that changes could be made to the image while making local features testable. Would we keep the `Dockerfile` in this repo? Can we default to using a local image over the remote one? I haven't done a whole lot of Docker, so there's probably a good answer. 😄